internal/clients/elasticsearch/index.go

```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
 	"github.com/elastic/terraform-provider-elasticstack/internal/models"
 	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	fwdiags "github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
```

Comment:
nit-pick: It seems the same file contains functions for both ILM (still SDKv2) and index (Framework) resources. Does it make sense to split them?

Reply:
This file has functions for:

* ILM
* Indices
* Templates
* Component templates
* Ingest processing
...

I'm happy to split it up, but I don't think it should be an SDK vs FW thing. WDYT?

Reply:
I mean that having one file per resource (e.g. indices) would allow us not to mix both SDK and FW during migration, FWIW (and besides other benefits). However, if we cannot split it like that then obviously there is no sense to do it based on SDK vs FW thing.